From b102865225c8db6a4ef60f346aded5959acc4d1c Mon Sep 17 00:00:00 2001
From: Marco Felsch <m.felsch@pengutronix.de>
Date: Wed, 9 Nov 2022 12:59:09 +0100
Subject: [PATCH 3/4] feat(build): add support for new binutils versions

The new ld.bfd linker version has added new warnings which are enabled by
default to secure elf binaries:
 - https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=ba951afb99912da01a6e8434126b8fac7aa75107
 - https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=0d38576a34ec64a1b4500c9277a8e9d0f07e6774

Since the tf-a handles every warning as error we need to tell the linker
to not warn about this. This is required to make the bl31 target
buildable with newer toolchain versions.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>
Change-Id: I9430f5fa5036ca88da46cd3b945754d62616b617
---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 3941f869800a..af7fd1535694 100644
--- a/Makefile
+++ b/Makefile
@@ -444,6 +444,10 @@ TF_LDFLAGS		+=	$(TF_LDFLAGS_$(ARCH))
 endif
 endif
 
+# With ld.bfd version 2.29 and newer new warnings are added. Skip those since we
+# are not loaded by a elf loader.
+TF_LDFLAGS		+=	$(call ld_option, --no-warn-rwx-segments)
+
 DTC_FLAGS		+=	-I dts -O dtb
 DTC_CPPFLAGS		+=	-P -nostdinc -Iinclude -Ifdts -undef \
 				-x assembler-with-cpp $(DEFINES)
-- 
2.30.2

